state/state.go

```diff
@@ -264,12 +273,26 @@ func (s *State) GetValidators() (*types.ValidatorSet, *types.ValidatorSet) {
 	return s.LastValidators, s.Validators
 }
 
+// ChainID returns the chain ID.
+func (s *State) ChainID() string {
+	return s.chainID
+}
+
+// SetChainID sets the chain ID.
+func (s *State) SetChainID(chainID string) {
+	s.chainID = chainID
```

Comment:
eh. Maybe it's fine if ChainID is written to disk every time - it's small, and might be worth avoiding this extra step of needing to set it. I can see folks easily forgetting to call this, and everything will probably break without it.